### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by aurya-dev**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -56,14 +56,14 @@ Purpose:  This project will show you the difference between member functions and
...
 };
 
-struct <#structName1#>                                //4
+struct struct1                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
```
Comment (matkatmusic) [inline main.cpp]: What do we always check before using pointers? 
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -73,29 +73,36 @@ struct <#structName1#>                                //4
...
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float name1 { 30 }, name2 { 20 };
+    float memberFunction(float* updatedValue)      //12
     {
-        
+        std::cout << "U's name1 value: " << name1 << std::endl;
+        name1 = *updatedValue;
+        std::cout << "U's name1 updated value: " << name1 << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: What do we always check before using pointers? 
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -73,29 +73,36 @@ struct <#structName1#>                                //4
...
+    static float staticFunctionA(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's name1 value: " << that->name1 << std::endl;
+        that->name1 = *updatedValue;
+        std::cout << "U's name1 updated value: " << that->name1 << std::endl;
+        while( std::abs(that->name2 - that->name1) > 0.001f )
```
Comment (matkatmusic) [inline main.cpp]: What do we always check before using pointers? 
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -115,19 +122,22 @@ struct <#structname2#>
...
-    T <#name2#>( , );                                             //6
+    T t1(10 , "a");                                             //6
+    T t2(20 , "b");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    struct1 f;                                            //7
+    auto* smaller = f.compare(&t1 , &t2);                              //8
+    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
```
Comment (matkatmusic) [inline main.cpp]: What do we always check before using pointers? 
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (aurya-dev) [reply main.cpp]: Always check to make sure your pointer is not NULL before using it!!!! :)

File: main.cpp (hunk 2 above)
Comment (aurya-dev) [reply main.cpp]: Always check to make sure your pointer is not NULL before using it!!!! :)

File: main.cpp (hunk 3 above)
Comment (aurya-dev) [reply main.cpp]: Always check to make sure your pointer is not NULL before using it!!!! :)

File: main.cpp (hunk 4 above)
Comment (aurya-dev) [reply main.cpp]: Always check to make sure your pointer is not NULL before using it!!!! :)

File: main.cpp [hunk 5]
```diff
@@ -56,46 +56,64 @@ Purpose:  This project will show you the difference between member functions and
...
+struct struct2
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunctionA(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(updatedValue != nullptr)
```
Comment (matkatmusic) [inline main.cpp]: `that` is a pointer.  You need to check if `that` is null before you use it.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 6]
```diff
@@ -56,46 +56,64 @@ Purpose:  This project will show you the difference between member functions and
...
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const std::string c) : value(v), name(c) {}   //1
+    int value;
+    std::string name; 
 };
 
-struct <#structName1#>                                //4
+struct struct1                                //4
```
Comment (matkatmusic) [inline main.cpp]: Type names need to begin with a capital letter.
- [x] revise

File: main.cpp [hunk 7]
```diff
@@ -56,46 +56,64 @@ Purpose:  This project will show you the difference between member functions and
...
+            }
+            std::cout << "U's name2 updated value: " << name2 << std::endl;
+            return name2 * name1;   
+        }
+        return 0;
     }
 };
 
-struct <#structname2#>
+struct struct2
```
Comment (matkatmusic) [inline main.cpp]: Type names need to begin with a capital letter.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**